service/index/writer.go

```diff
@@ -26,46 +27,46 @@ import (
 // Writer implements the `index.Writer` interface to write indexing data to
 // an underlying Badger database.
 type Writer struct {
-	db      *badger.DB
-	storage Storage
+	db  *badger.DB
+	lib dps.WriteLibrary
```

Comment:
I really don't like this. When reading the usage of this attribute:

```go
w.lib.SaveLast(height)
```

It's really not obvious to me what `lib` could be, without knowing that it's a struct in our storage package that we use to inject our codec. I really prefer the previous naming of `storage`, this way is much clearer IMO.

Also I'm not fond of `WriteLibrary` compared to `StorageLibrary`. Write could mean many other things, like writing in `Stdout` while `Storage` is explicit about storing something so it implies persistence, which is what we're aiming for here with our `index` package.

Reply:
Well, I really disagree, because `storage` implies the component will actually store something. When, in truth, all it provides is a library of storage functions. I made the change her to be consistent with the interface naming, which I didn't change, by the way. All I did was change to the smaller interface, as it didn't really need the whole interface.

Reply:
I kinda agree with @Ullaakut here, naming isn't that perfect. I can see why the change was made though.

We could embed the type and then have `w.SaveLast(height)`, though I'm not sure how cleanly it would propagate through the code for all usages..

Reply:
You would still have `w.db.Update(w.SaveLast(height))`.